contracts/test/forkMainnet/L2Deposit/Deposit.t.sol

```diff
@@ -43,9 +53,18 @@ contract TestL2DepositTopUp is TestL2Deposit {
         // overwrite deposit data with encoded arbitrum specific params
         DEFAULT_DEPOSIT.data = abi.encode(arbMaxSubmissionCost, arbMaxGas, arbGasPriceBid);
 
-        // compose payload with signature
-        bytes memory sig = _signDeposit(userPrivateKey, DEFAULT_DEPOSIT);
-        bytes memory payload = abi.encodeWithSelector(L2Deposit.deposit.selector, IL2Deposit.DepositParams(DEFAULT_DEPOSIT, sig));
+        // sign the deposit action
+        bytes memory depositActionSig = _signDeposit(userPrivateKey, DEFAULT_DEPOSIT);
+        // create spendWithPermit using the deposit and sign it
+        // _createSpenderPermitFromL2Deposit() will create EIP712 hash of the deposit for us
```

Comment:
This detail should probably be included in `_createSpenderPermitFromL2Deposit` since the hash is not used here.